### PR TITLE
add tidbctl and showTableID command

### PR DESCRIPTION
**Pull request by LiangShang**
add tidbctl and showTableID command, which prints the corresponding id given the table name

### CODE REVIEW

File: tidbctl/command/table_id_command.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	return m
+}
+
+func showTableIDCommandFunc(cmd *cobra.Command, args []string) {
+	if len(args) != 3 {
+		fmt.Println("Usage: showTableID <tidb_addr> <db_name> <table_name>")
+		return
+
+	}
+	host := args[0]
```
Comment (siddontang) [inline tidbctl/command/table_id_command.go]: I think we must pass TiDB address to the tidb-ctl at first, not in the sub command.

File: tidbctl/command/table_id_command.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	err = json.Unmarshal(content, &ti)
+	if err != nil {
+		fmt.Println("Error when unmarshaling the response body", string(content))
+		fmt.Println("The error is", err)
+		return
+	}
+	if verbose {
+		fmt.Println(ti)
+	} else {
+		fmt.Println("TableID:", ti.ID)
```
Comment (siddontang) [inline tidbctl/command/table_id_command.go]: I  prefer add a blank space after the semicolon.

File: tidbctl/command/table_id_command.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	}
+
+	var ti tableInfo
+	err = json.Unmarshal(content, &ti)
+	if err != nil {
+		fmt.Println("Error when unmarshaling the response body", string(content))
+		fmt.Println("The error is", err)
+		return
+	}
+	if verbose {
```
Comment (siddontang) [inline tidbctl/command/table_id_command.go]: I think no matter the verbose is true or false, we should output table ID independently below.

Comment (LiangShang) [reply tidbctl/command/table_id_command.go]: done

**Comment by siddontang** [comment]
LGTM

PTAL @disksing

**Comment by disksing** [comment]
@tiancaiamao Please help to find out what's going on with the CI.

File: cmd/tidb-ctl/main.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
+// Copyright 2016 PingCAP, Inc.
```
[the remark this answers is not in the document]
Comment (LiangShang) [reply cmd/tidb-ctl/main.go]: fixed

File: cmd/tidb-ctl/main.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	"io"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/pingcap/tidb/tidbctl"
```
[the remark this answers is not in the document]
Comment (LiangShang) [reply cmd/tidb-ctl/main.go]: fixed

File: cmd/tidb-ctl/main.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+	var input []string
+	stat, _ := os.Stdin.Stat()
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
+		detach = true
+		b, err := ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		input = strings.Split(strings.TrimSpace(string(b[:])), " ")
```
[the remark this answers is not in the document]
Comment (LiangShang) [reply cmd/tidb-ctl/main.go]: fixed

File: tidbctl/command/table_id_command.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+	}
+
+	resp, err := http.Get(urlString)
+
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	// fmt.Println(resp)
```
[the remark this answers is not in the document]
Comment (LiangShang) [reply tidbctl/command/table_id_command.go]: done

File: tidbctl/ctl.go [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tidbctl
+
+import (
+	"fmt"
+
+	"errors"
```
Comment (disksing) [inline tidbctl/ctl.go]: We use `github.com/juju/errors` instead of `errors` package.

**Comment by shenli** [comment]
@LiangShang We use "github.com/juju/errors" instead of standard library "errors".

**Comment by disksing** [comment]
Duplicated with #4880

**Pull request closed without merging**